packages/react-components/src/components/chart/legend/useChartsLegendTanstack.tsx

```diff
@@ -0,0 +1,244 @@
+  const hiddenDataStreams = useChartStore((state) => state.hiddenDataStreams);
+  const isDataStreamHidden = isDataStreamInList(hiddenDataStreams);
+  const propertyVisibilityIcon = isDataStreamHidden(datastream) ? (
+    <img alt='hide property' src={Hide}></img>
+  ) : (
+    <img alt='show property' src={Show}></img>
+  );
+
+  const toggleVisibility = () => {
+    if (isDataStreamHidden(datastream)) {
+      unHideDataStream(datastream);
+    } else {
+      if (isDataStreamHighlighted(datastream)) {
+        unHighlightDataStream(datastream);
+      }
+      hideDataStream(datastream);
+    }
+  };
+
+  const [lineIcon] = useHover((isHovering) => (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+    <div
+      style={{
+        backgroundColor: isHovering ? colorBackgroundDropdownItemHover : 'white', //white background color matches surroundings to display that it is diabled
+        borderRadius: borderRadiusDropdown,
+      }}
+      className={`base-chart-legend-row-line-container ${isDataStreamHidden(datastream) ? 'disabled' : ''}`}
+      onClick={toggleHighlighted}
+      aria-disabled={isDataStreamHidden(datastream)}
+      title={
+        isDataStreamHighlighted(datastream)
+          ? `Un-Highlight ${datastream.name} Proprerty`
+          : `Highlight ${datastream.name} Property`
+      }
+      role='button'
+      tabIndex={isDataStreamHidden(datastream) ? -1 : 0}
+    >
+      <div
+        className={`base-chart-legend-row-line-ind ${
+          isDataStreamHighlighted(datastream) ? 'base-chart-legend-row-line-ind-highlighted' : ''
+        }`}
+        style={{
+          backgroundColor: lineColor,
+        }}
+      />
+    </div>
+  ));
+
+  const [hideShowButton] = useHover((isHovering) => (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+    <div
+      style={{
+        backgroundColor: isHovering ? colorBackgroundDropdownItemHover : undefined,
+        borderRadius: borderRadiusDropdown,
+      }}
+      className='base-chart-legend-row-line-container'
+    >
+      <div
+        title={
+          isDataStreamHidden(datastream) ? `Show ${datastream.name} Proprerty` : `Hide ${datastream.name} Property`
+        }
+      >
+        <Button onClick={toggleVisibility} variant='icon' iconSvg={propertyVisibilityIcon} />
+      </div>
+    </div>
+  ));
+
+  return (
+    <div className='base-chart-legend-row-data-container'>
+      {hideShowButton}
+      {lineIcon}
+      <div
+        className={`base-chart-legend-row-data ${isDataStreamHidden(datastream) ? 'hidden-legend' : ''}`}
+        style={{
+          marginBlock: spaceStaticXxs,
+          minWidth: `${width / LEGEND_NAME_MIN_WIDTH_FACTOR}px`,
+        }}
+        ref={nameRef}
+        title={isNameTruncated ? name : undefined}
+      >
+        {name}
+      </div>
+    </div>
+  );
+};
+
+const useChartsLegendTanstack = ({
```

Comment:
may be worth adding an exit condition like 

if(!enabed) return { columnDefinitions: [], items: []}

Reply:
updated.